src/build/index.js

```diff
@@ -33,13 +33,25 @@ module.exports = async (argv) => {
     env: {
       NODE_ENV: process.env.NODE_ENV || 'production',
       AEGIR_BUILD_ANALYZE: argv.bundlesize,
-      AEGIR_NODE: argv.node
+      AEGIR_NODE: argv.node,
+      AEGIR_TS: argv.ts
     },
     localDir: path.join(__dirname, '../..'),
     preferLocal: true,
     stdio: 'inherit'
   })
 
+  if (argv.ts) {
+    await execa('tsc', [
+      '--outDir', './dist/src',
```

Comment:
Is it possible to remove this so it doesn't override settings in tsconfig.json file?

Reply:
This only overrides when you run `aegir build --ts`, you can `tsc` with whatever is in tsconfig.json.
Here the outDir should be hardcoded, for consistency across repos, plus we clean `dist` before build so we want this inside the dist folder.
Any specific use case that you have in mind ?

Reply:
I would prefer ./dist instead of ./dist/src since if you try to import specific path you would have `import {} from "somelib/dist/src/someDir/moduleFile"`. If it's for compatibility with other stuff like webpack build, maybe it can be specified in aegir config?

It's just nitpick :slightly_smiling_face:

Reply:
humm its just because in dist we already have the script tag bundle, and if we compile to the same folder we will have src files mixed with the bundle.
thoughts ?

Reply:
BTW this PR includes initial work to support cosmiconfig, i will probably remove that from this PR and continue in another, but the goal is to overhaul the config to be easier to extend and allow for more config options.

Reply:
yeah, true, it makes no sense unless you move bundle into another directory but that's breaking change!
I just checked other libp2p repo, it's same convention so we should probably stick to it

Reply:
@hugomrdias I realize this already got merged, perhaps the time for discussion is over.
But IMO it seems strange to use a different outDir than `src/`. The reason is that aegir-maintained libp2p repos expect javascript source files to be in `src`, and submodules are sometimes directly imported like eg: `require('libp2p-xxx/src/submodule')`). If typescript aegir-maintained projects are to maximally maintain compatibility with non-typescript aegir-maintained projects, it seems that we would want to maintain this?

Reply:
Can you open a issue to discuss this please ?